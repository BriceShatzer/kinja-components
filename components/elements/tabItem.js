/* @flow */

import * as React from 'react';
import styled, { css } from 'styled-components';

const TabListElement = styled.li``;

export const TabLink = styled.a`
	&& {
		padding: 5px;
		text-decoration: none;
		color: ${props => props.theme.color.darksmoke};

		&:focus,
		&:hover {
			outline: none;
			color: ${props => props.theme.color.darksmoke};
		}

		${props => props.selected && css`
			color: ${props => props.theme.color.black};
			font-weight: bold;
			padding: 5px 3px;
		`}
	}
`;

type Props = {
	selected?: boolean,
	onClick?: () => void,
	value: string,
	label: string,
	index: number,
	hidden?: boolean,
	count?: number,
	className?: string
};

class TabItem extends React.PureComponent<Props> {
	render() {
		const {
			selected,
			onClick,
			value,
			label,
			index,
			count
		} = this.props;

		const hidden = this.props.hidden || false;
		const counter = count ? ` (${count})` : null;

		if (hidden) {
			return null;
		}

		return (
			<TabListElement className={this.props.className}>
				<TabLink
					onClick={onClick}
					role="link" tabIndex={index}
					value={value}
					selected={selected}
				>
					{label}{counter}
				</TabLink>
			</TabListElement>
		);
	}
}

export default TabItem;
