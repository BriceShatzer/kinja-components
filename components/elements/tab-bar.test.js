import * as React from 'react';
import { shallow, mount } from 'enzyme';

import TabBar from './tab-bar';
import { TabLink } from './tabItem';


describe('Tabs', () => {
	const onChange = jest.fn();

	const stubElement = () => (
		<TabBar activeTab="Music" onChange={onChange} tabs={['Music', 'Video', 'Text', 'Image']} />
	);

	it('should render properly', () => {
		const tabBar = shallow(stubElement());
		expect(tabBar).toMatchSnapshot();
	});

	it('should call onChange', () => {
		const tabBar = mount(stubElement());
		const tabItem = tabBar.find(TabLink).at(2);
		tabItem.simulate('click');

		expect(onChange).toHaveBeenCalledTimes(1);
		expect(onChange).toHaveBeenLastCalledWith('Text');
		onChange.mockClear();
	});
});
