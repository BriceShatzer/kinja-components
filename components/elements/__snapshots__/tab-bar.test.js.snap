// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should render properly 1`] = `
<EnsureDefaultTheme>
  <tab-bar__TabsContainer>
    <tab-bar__TabList
      onChange={[MockFunction]}
    >
      <tab-bar__StyledTabItem
        index={0}
        key="Music"
        label="Music"
        selected={true}
        value="Music"
      />
      <tab-bar__StyledTabItem
        index={1}
        key="Video"
        label="Video"
        selected={false}
        value="Video"
      />
      <tab-bar__StyledTabItem
        index={2}
        key="Text"
        label="Text"
        selected={false}
        value="Text"
      />
      <tab-bar__StyledTabItem
        index={3}
        key="Image"
        label="Image"
        selected={false}
        value="Image"
      />
    </tab-bar__TabList>
  </tab-bar__TabsContainer>
</EnsureDefaultTheme>
`;
